File: data-structures/list/interfaces/list.ts
export interface List<T> {
  length(): number;
  capacity(): number;
  isEmpty(): boolean;
  get(index: number): T | undefined;
  set(index: number, value: T): void;
  append(value: T): void;
  prepend(value: T): void;
  insert(index: number, value: T): void;
  removeAt(index: number): T | undefined;
  remove(value: T): boolean;
  contains(value: T): boolean;
  indexOf(value: T): number;
  clear(): void;
  toArray(): T[];
}

File: data-structures/list/interfaces/list.go
package interfaces

type List[T comparable] interface {
	Length() int
	Capacity() int
	IsEmpty() bool
	Get(index int) T
	Set(index int, value T)
	Append(value T)
	Prepend(value T)
	Insert(index int, value T)
	RemoveAt(index int) T
	Remove(value T) bool
	Contains(value T) bool
	IndexOf(value T) int
	Clear()
	ToSlice() []T
}
